unified-prototype/unified-plugin/plugin-android/src/main/java/org/gradle/api/experimental/android/AndroidLibrary.java

```diff
@@ -2,65 +2,75 @@
 
 import com.android.build.api.dsl.BaseFlavor;
 import com.android.build.api.dsl.CommonExtension;
+import com.h0tk3y.kotlin.staticObjectNotation.Configuring;
+import com.h0tk3y.kotlin.staticObjectNotation.Restricted;
 import org.gradle.api.Action;
-import org.gradle.api.NamedDomainObjectContainer;
 import org.gradle.api.experimental.common.LibraryDependencies;
+import org.gradle.api.model.ObjectFactory;
 import org.gradle.api.provider.Property;
-import org.gradle.api.tasks.Input;
-import org.gradle.api.tasks.Nested;
+
+import javax.inject.Inject;
 
 /**
  * The public DSL interface for a declarative Android library.
  */
+@Restricted
 public abstract class AndroidLibrary {
+    private final LibraryDependencies dependencies;
     private final AndroidTargets targets;
 
-    public AndroidLibrary(AndroidTargets targets) {
+    @Inject
+    public AndroidLibrary(AndroidTargets targets, ObjectFactory objectFactory) {
         this.targets = targets;
+        this.dependencies = objectFactory.newInstance(LibraryDependencies.class);
     }
 
     /**
      * @see CommonExtension#getNamespace()
      */
-    @Input
+    @Restricted
     public abstract Property<String> getNamespace();
 
     /**
      * @see CommonExtension#getCompileSdk()
      */
-    @Input
+    @Restricted
     public abstract Property<Integer> getCompileSdk();
 
     /**
      * @see BaseFlavor#getMinSdk()
      */
-    @Input
+    @Restricted
     public abstract Property<Integer> getMinSdk();
 
     /**
      * JDK version to use for compilation.
      */
-    @Input
+    @Restricted
     public abstract Property<Integer> getJdkVersion();
 
     /**
      * Common dependencies for all targets.
      */
-    @Nested
-    public abstract LibraryDependencies getDependencies();
+    @Restricted
+    public LibraryDependencies getDependencies() {
```

Comment:
I'm assuming the declarative parser can't handle nested properties like this?

Reply:
Hmm, actually, it looks like `@Nested` works fine.  I changed this in the course of getting it to work and didn't change it back.  I'll do that now.